### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by 6ALLIKA**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -16,6 +16,32 @@ public Rectangle(Integer width, Integer length, String color) {
...
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null) {
+            return false;
+        }
+        if (o.getClass().equals(Rectangle.class)) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        if (o.getClass().equals(getClass())) {
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -16,6 +16,32 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 7;
+        result = 17 * result + color.hashCode();
+        result = 17 * result + width;
+        result = 17 * result + length;
+        return result;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: what if field would be null?

Comment (6ALLIKA) [reply src/main/java/core/basesyntax/Rectangle.java]: good call, it would be NPE

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -16,6 +16,32 @@ public Rectangle(Integer width, Integer length, String color) {
...
+                    && color.equals(rectangle.color);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 7;
+        result = (color == null) ? 17 * result : 17 * result + color.hashCode();
+        result = (width == null) ? 17 * result : 17 * result + width;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        result = (width == null) ? 17 * result : 17 * result + width.hashCode();
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -16,6 +16,32 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        result = (length == null) ? 17 * result : 17 * result + length;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        result = (length == null) ? 17 * result : 17 * result + length.hashCode();
```

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**